akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -1064,7 +1116,33 @@ object GraphDSL extends GraphApply {
       source.shape.out
     }
 
-    private[stream] def traversalBuilder: TraversalBuilder = traversalBuilderInProgress
+    private[GraphDSL] def traversalBuilder: TraversalBuilder = traversalBuilderInProgress
+
+    private[stream] def result(resultShape: Shape): TraversalBuilder = {
+      def errorString[T](expectedSet: Set[T], actualSet: Set[T], tag: String)(format: T ⇒ String): String =
+        if (expectedSet != actualSet) {
+          val diff1 = expectedSet.diff(actualSet)
+          val diff2 = actualSet.diff(expectedSet)
+
+          val forwardMessage =
+            if (diff1.isEmpty) ""
+            else s" $tag ${diff1.map(format).mkString(", ")} were returned in the resulting shape but were already connected."
+
+          val backwardMessage =
+            if (diff2.isEmpty) ""
+            else s" $tag ${diff2.map(format).mkString(", ")} were not returned in the resulting shape but also not connected."
```

Comment:
Funky english? "were not returned in the resulting shape _and_ also not connected"?